camel/loaders/panda_reader.py

```diff
@@ -0,0 +1,328 @@
+        file_path = str(data)
+        path = Path(file_path)
+        if not file_path.startswith("http") and not path.exists():
+            raise FileNotFoundError(f"File {file_path} not found")
+        if path.suffix == ".csv":
+            return SmartDataframe(
+                self.read_csv(file_path, *args, **kwargs), config=self.config
+            )
+        elif path.suffix == ".xlsx" or path.suffix == ".xls":
+            return SmartDataframe(
+                self.read_excel(file_path, *args, **kwargs), config=self.config
+            )
+        elif path.suffix == ".json":
+            return SmartDataframe(
+                self.read_json(file_path, *args, **kwargs), config=self.config
+            )
+        elif path.suffix == ".parquet":
+            return SmartDataframe(
+                self.read_parquet(file_path, *args, **kwargs),
+                config=self.config,
+            )
+        elif path.suffix == ".sql":
+            return SmartDataframe(
+                self.read_sql(file_path, *args, **kwargs), config=self.config
+            )
+        elif path.suffix == ".html":
+            return SmartDataframe(
+                self.read_html(file_path, *args, **kwargs), config=self.config
+            )
+        elif path.suffix == ".feather":
+            return SmartDataframe(
+                self.read_feather(file_path, *args, **kwargs),
+                config=self.config,
+            )
+        elif path.suffix == ".dta":
+            return SmartDataframe(
+                self.read_stata(file_path, *args, **kwargs), config=self.config
+            )
+        elif path.suffix == ".sas":
+            return SmartDataframe(
+                self.read_sas(file_path, *args, **kwargs), config=self.config
+            )
+        elif path.suffix == ".pkl":
+            return SmartDataframe(
+                self.read_pickle(file_path, *args, **kwargs),
+                config=self.config,
+            )
+        elif path.suffix == ".h5":
+            return SmartDataframe(
+                self.read_hdf(file_path, *args, **kwargs), config=self.config
+            )
+        elif path.suffix == ".orc":
+            return SmartDataframe(
+                self.read_orc(file_path, *args, **kwargs), config=self.config
+            )
+        else:
+            raise ValueError(f"Unsupported file format: {path.suffix}")
+
+    def read_csv(
+        self, file_path: str, *args: Any, **kwargs: Dict[str, Any]
+    ) -> DataFrame:
+        r"""Reads a CSV file and returns a DataFrame.
+
+        Args:
+            file_path (str): The path to the CSV file.
+            *args (Any): Additional positional arguments.
+            **kwargs (Dict[str, Any]): Additional keyword arguments.
+
+        Returns:
+            DataFrame: The DataFrame object.
+        """
+        if not file_path.endswith('.csv'):
+            raise ValueError("Only CSV files are supported")
```

Comment:
We can probably use a decorator to do this checking to eliminate those similar checking codes